rerun_py/src/python_bridge.rs

```diff
@@ -362,108 +361,78 @@ fn serve(open_browser: bool, web_port: Option<u16>, ws_port: Option<u16>) -> PyR
-
-    #[cfg(not(feature = "web_viewer"))]
-    {
-        _ = port;
-        Err(PyRuntimeError::new_err(
-            "The Rerun SDK was not compiled with the 'web_viewer' feature",
-        ))
-    }
-}
-
-#[pyfunction]
-fn get_app_url() -> String {
-    let session = python_session();
-    session.get_app_url()
-}
-
-#[pyfunction]
-fn shutdown(py: Python<'_>) {
-    re_log::debug!("Shutting down the Rerun SDK");
-    // Disconnect the current sink which ensures that
-    // it flushes and cleans up.
-    disconnect(py);
-}
-
-/// Is logging enabled in the global session?
-#[pyfunction]
-fn is_enabled() -> bool {
-    python_session().is_enabled()
-}
-
-/// Enable or disable logging in the global session.
+/// Disconnect from remote server (if any).
 ///
-/// This is a global setting that affects all threads.
-/// By default logging is enabled.
-#[pyfunction]
-fn set_enabled(enabled: bool) {
-    python_session().set_enabled(enabled);
-}
-
-/// Block until outstanding data has been flushed to the sink
+/// Subsequent log messages will be buffered and either sent on the next call to `connect`,
+/// or shown with `show`.
 #[pyfunction]
-fn flush(py: Python<'_>) {
+fn disconnect(py: Python<'_>) {
     // Release the GIL in case any flushing behavior needs to
     // cleanup a python object.
     py.allow_threads(|| {
-        python_session().flush();
+        let rec_stream = global_data_stream();
+        let Some(rec_stream) = rec_stream.as_ref() else {
+            no_active_recording("disconnect");
+            return;
+        };
+        rec_stream.disconnect();
     });
 }
 
-/// Disconnect from remote server (if any).
-///
-/// Subsequent log messages will be buffered and either sent on the next call to `connect`,
-/// or shown with `show`.
+// TODO
+/// Block until outstanding data has been flushed to the sink
 #[pyfunction]
-fn disconnect(py: Python<'_>) {
+fn flush(py: Python<'_>) {
     // Release the GIL in case any flushing behavior needs to
     // cleanup a python object.
     py.allow_threads(|| {
-        python_session().disconnect();
+        let rec_stream = global_data_stream();
+        let Some(rec_stream) = rec_stream.as_ref() else {
+            no_active_recording("flush");
+            return;
+        };
+        rec_stream.flush_blocking();
     });
 }
 
 #[pyfunction]
-fn save(path: &str) -> PyResult<()> {
-    let mut session = python_session();
-    session
-        .save(path)
-        .map_err(|err| PyRuntimeError::new_err(err.to_string()))
-}
-
-#[pyclass]
-struct PyMemorySinkStorage(MemorySinkStorage);
-
-#[pymethods]
-impl PyMemorySinkStorage {
-    fn get_rrd_as_bytes<'p>(&self, py: Python<'p>) -> PyResult<&'p PyBytes> {
-        self.0
-            .rrd_as_bytes()
-            .map(|bytes| PyBytes::new(py, bytes.as_slice()))
-            .map_err(|err| PyRuntimeError::new_err(err.to_string()))
-    }
-}
+// TODO(jleibs) expose this as a python type
+fn start_web_viewer_server(port: u16) -> PyResult<()> {
```

Comment:
You might want to cherry-pick this over as a starting point:
https://github.com/rerun-io/rerun/pull/2010/commits/707e76c76c35d111bef9e0e5192a5644d6d5da7e